types/ember/test/ember-module-tests.ts

```diff
@@ -0,0 +1,280 @@
+Ember.addListener({ a: 'foo' }, 'a', {}, () => {});
+Ember.addListener({ a: 'foo' }, 'a', null, () => {});
+// addObserver
+Ember.addObserver({ a: 'foo' }, 'a', null, () => {});
+Ember.addObserver({ a: 'foo' }, 'a', {}, () => {});
+// aliasMethod
+Ember.aliasMethod('init');
+// assert
+Ember.assert('2+2 should always be 4', 2 + 2 === 4);
+// assign
+const o1 = Ember.assign({ a: 1 }, { b: 2 });
+o1.a; // $ExpectType number
+o1.b; // $ExpectType number
+o1.c; // $ExpectError
+// Ember.bind // $ExpectError
+// cacheFor
+Ember.cacheFor({ a: 123 }, 'a'); // $ExpectType number | undefined
+Ember.cacheFor({ a: 123 }, 'x'); // $ExpectError
+// compare
+Ember.compare('31', '114'); // $ExpectType number
+// copy
+Ember.copy({ a: 12 }, true).a; // $ExpectType number
+Ember.copy({ a: 12 }); // $ExpectType any
+Ember.copy({ a: 12 }).a; // $ExpectType any
+Ember.copy({ a: 12 }).b; // $ExpectType any
+// debug
+Ember.debug('some info for developers');
+// deprecate
+Ember.deprecate("you shouldn't use this anymore", 3 === 3, {
+    id: 'no-longer-allowed',
+    until: '99.0.0'
+});
+// get
+Ember.get({ z: 23 }, 'z'); // $ExpectType number
+Ember.get({ z: 23 }, 'zz'); // $ExpectError
+// getEngineParent
+Ember.getEngineParent(new EngineInstance()); // $ExpectType EngineInstance
+// getOwner
+Ember.getOwner(new Component());
+// getProperties
+Ember.getProperties({ z: 23 }, 'z').z; // $ExpectType number
+Ember.getProperties({ z: 23 }, 'z', 'z').z; // $ExpectType number
+Ember.getProperties({ z: 23 }, 'z', 'a').z; // $ExpectError
+Ember.getProperties({ z: 23 }, ['z', 'z']).z; // $ExpectType number
+Ember.getProperties({ z: 23 }, ['z', 'a']).z; // $ExpectError
+// getWithDefault
+Ember.getWithDefault({ z: 23 }, 'z', 43); // $ExpectType number
+Ember.getWithDefault({ a: undefined as number | undefined, z: 23 }, 'a', 99); // $ExpectType number | undefined
+
+// guidFor
+Ember.guidFor({}); // $ExpectType string
+Ember.guidFor(''); // $ExpectType string
+// isArray
+Ember.isArray(''); // $ExpectType boolean
+Ember.isArray([]); // $ExpectType boolean
+// isBlank
+Ember.isBlank(''); // $ExpectType boolean
+Ember.isBlank([]); // $ExpectType boolean
+// isEmpty
+Ember.isEmpty(''); // $ExpectType boolean
+Ember.isEmpty([]); // $ExpectType boolean
+// isEqual
+Ember.isEqual('', 'foo'); // $ExpectType boolean
+Ember.isEqual([], ''); // $ExpectType boolean
+// isNone
+Ember.isNone(''); // $ExpectType boolean
+Ember.isNone([]); // $ExpectType boolean
+// isPresent
+Ember.isPresent(''); // $ExpectType boolean
+Ember.isPresent([]); // $ExpectType boolean
+// merge
+Ember.merge({ a: 12 }, { b: 34 }).a; // $ExpectType number
+Ember.merge({}, {});
+// observer
+const o2 = EmberObject.extend({
+    name: 'foo',
+    nameWatcher: Ember.observer('name', () => {}),
+    nameWatcher2: Ember.observer('name', 'fullName', () => {})
+});
+// on
+const o3 = EmberObject.extend({
+    name: 'foo',
+    nameWatcher: Ember.on('init', () => {}),
+    nameWatcher2: Ember.on('destroy', () => {})
```

Comment:
2nd one should have two events listed?

Reply:
This is a smoke test -- I'm not trying to exhaustively test the whole API. See other tests for deeper dives into these areas